charts/seed-monitoring/charts/core/charts/prometheus/templates/allow-prometheus-network-policy.yaml

```diff
@@ -17,6 +17,12 @@ spec:
     - podSelector:
         matchLabels:
           networking.gardener.cloud/from-prometheus: allowed
+    - podSelector:
+        matchLabels:
+          networking.gardener.cloud/from-prometheus: allowed
+      namespaceSelector:
+        matchLabels:
+          role: garden
```

Comment:
Does it make sense to make this pod selector more specific to only allow talking to the kube-state-metrics deployment (i.e. using `component: kube-state-metrics`)? We could also use a dedicated label for the direction `shoot prometheus -> seed component` to distinguish from `seed prometheus -> seed component` (e.g. `networking.gardener.cloud/from-shoot-prometheus`).
We probably don't want to allow the Shoot prometheus' (I don't remember the correct plural 😉) to talk to all seed components that want to be scraped by the seed prometheus.

Reply:
Makes sense. I can make the label more specific to ksm.

Regarding the dedicated label, this would make sense if we start scraping multiple components in the garden ns, but this is not (yet) the case. If there is a use case where we need to scrape more central components, then I would suggest we add a dedicated label.